### PR TITLE
NextTo relation handles all sides

### PR DESCRIPTION
**Pull request by cvolkcvolk**
## Summary
Extend the `NextToLossStrategy` to support all four placement sides (FRONT, BACK, LEFT, RIGHT) and fixes a bug where child objects could extend beyond the parent's extent. 

Misc: 
- Extended `isaac_sim_object_placer_notebook.py` & `dummy_object_placer_notebook.py` to demonstrate all four sides with a arrangement around a central object.

### CODE REVIEW

File: isaaclab_arena/relations/relation_loss_strategies.py [hunk 1]
```diff
@@ -20,6 +21,33 @@
...
+class SideConfig:
+    """Configuration for computing NextTo loss for a given side.
+
+    Attributes:
+        primary_axis: Index of the axis along which child is placed (0=X, 1=Y).
+        direction: +1 if child should be in positive direction from parent (RIGHT, BACK),
+                   -1 if child should be in negative direction (LEFT, FRONT).
+    """
+
+    primary_axis: int  # 0=X, 1=Y
```
Comment (xyao-nv) [inline isaaclab_arena/relations/relation_loss_strategies.py]: I wonder if Enum would be a better choice to establish those mappings easily.

File: isaaclab_arena_examples/relations/isaac_sim_object_placer_notebook.py [hunk 2]
```diff
@@ -48,34 +46,60 @@ def run_isaac_sim_object_placer_demo(num_steps: int = 10000):
...
+        ground_plane,
+        light,
+        office_table,
+        cracker_box,
+        dex_cube,
+        sugar_box,
+        mustard_bottle,
+        tomato_soup_can,
+        mug,
+    ]
```
Comment (xyao-nv) [inline isaaclab_arena_examples/relations/isaac_sim_object_placer_notebook.py:98]: Can't wait for the day when it can place all objs automatically from obj_library.py & LW SDK into a kitchen scene 😆

File: isaaclab_arena/relations/relation_loss_strategies.py [hunk 3]
```diff
@@ -89,57 +117,69 @@ def compute_loss(
...
-        parent_y_max = parent_pos[1] + parent_bbox.max_point[1]  # Back edge
-
-        # 1. Half-plane loss: child must be to the right of parent's right edge
-        right_side_loss = single_boundary_linear_loss(
-            child_pos[0],
-            parent_x_max,
+        # Select parent edge and child offset based on direction
+        # direction +1 (RIGHT/BACK): child in positive direction, use parent max edge
+        # direction -1 (LEFT/FRONT): child in negative direction, use parent min edge
+        if cfg.direction == +1:
```
Comment (xyao-nv) [inline isaaclab_arena/relations/relation_loss_strategies.py]: Maybe it's me with bad memories, is it possible to not let users be bothered by the relationship between +1 and right/back while understanding this code?

File: isaaclab_arena/relations/relation_loss_strategies.py [hunk 4]
```diff
@@ -89,57 +117,69 @@ def compute_loss(
...
-        # 3. Distance loss: child's left edge should be at distance from parent's right edge
-        assert distance >= 0.0, f"NextTo distance must be non-negative, got {distance}"
-        # child_x_min = child_pos[0] + child_bbox.min_point[0], so:
-        # target: child_pos[0] + child_bbox.min_point[0] = parent_x_max + distance
-        # target_child_pos_x = parent_x_max + distance - child_bbox.min_point[0]
-        target_x = parent_x_max + distance - child_bbox.min_point[0]
-        next_to_distance_loss = single_point_linear_loss(child_pos[0], target_x, slope=self.slope)
+        # 3. Distance loss: child edge at target distance from parent edge
+        # For direction +1: target = parent_max + distance - child_min
+        # For direction -1: target = parent_min - distance - child_max
```
Comment (xyao-nv) [inline isaaclab_arena/relations/relation_loss_strategies.py:173]: If you switch to Enum, then you can add a linear ops func to map 0/1 to +1/-1.

File: isaaclab_arena/relations/relation_loss_strategies.py [hunk 5]
```diff
@@ -20,6 +21,33 @@
...
+    direction: int  # +1 or -1
+
+    @property
+    def band_axis(self) -> int:
+        """Perpendicular axis for band constraint."""
+        return 1 - self.primary_axis
+
+
+SIDE_CONFIGS: dict[Side, SideConfig] = {
+    Side.RIGHT: SideConfig(primary_axis=0, direction=+1),
```
Comment (alexmillane) [inline isaaclab_arena/relations/relation_loss_strategies.py]: Just to echo @xyao-nv 's comment. I think if you define Enums these might look a bit more readable

```
Side.RIGHT: SideConfig(primary_axis=Axis.X, direction=Direction.Positive),
```

File: isaaclab_arena/relations/relation_loss_strategies.py [hunk 6]
```diff
@@ -20,6 +21,33 @@
...
+    """Configuration for computing NextTo loss for a given side.
+
+    Attributes:
+        primary_axis: Index of the axis along which child is placed (0=X, 1=Y).
+        direction: +1 if child should be in positive direction from parent (RIGHT, BACK),
+                   -1 if child should be in negative direction (LEFT, FRONT).
+    """
+
+    primary_axis: int  # 0=X, 1=Y
+    direction: int  # +1 or -1
```
Comment (alexmillane) [inline isaaclab_arena/relations/relation_loss_strategies.py]: see @xyao-nv 's comment above.

File: isaaclab_arena/relations/relation_loss_strategies.py (hunk 3 above)
Comment (alexmillane) [reply isaaclab_arena/relations/relation_loss_strategies.py]: Potentially if we use the Enums as suggested above, this might look more understandable.

File: isaaclab_arena_examples/relations/isaac_sim_object_placer_notebook.py (hunk 2 above)
Comment (alexmillane) [reply isaaclab_arena_examples/relations/isaac_sim_object_placer_notebook.py:98]: That day is on the way!

File: isaaclab_arena/relations/relation_loss_strategies.py (hunk 1 above)
Comment (cvolkcvolk) [reply isaaclab_arena/relations/relation_loss_strategies.py]: Done. Implemented an Axis and Direction Enum.

File: isaaclab_arena/relations/relation_loss_strategies.py (hunk 4 above)
Comment (cvolkcvolk) [reply isaaclab_arena/relations/relation_loss_strategies.py:173]: That's a good idea!

File: isaaclab_arena/relations/relation_loss_strategies.py (hunk 5 above)
Comment (cvolkcvolk) [reply isaaclab_arena/relations/relation_loss_strategies.py]: Thanks for the hint! Introduced an `Axis` and `Direction` enum with which the above config becomes much cleaner
